src/map/utils/battleutils.cpp

```diff
@@ -6984,73 +6951,16 @@ namespace battleutils
 
             if (PCoverAbilityUser != nullptr)
             {
-                float coverAbilityTargetX = PCoverAbilityTarget->loc.p.x;
-                float coverAbilityTargetZ = PCoverAbilityTarget->loc.p.z;
-                float mobX                = PMob->loc.p.x;
-                float mobZ                = PMob->loc.p.z;
-
-                float xdif       = coverAbilityTargetX - mobX;
-                float zdif       = coverAbilityTargetZ - mobZ;
-                float slope      = 0;
-                float maxSlope   = 0;
-                float minSlope   = 0;
-                bool  zDependent = true; // using a slope where z is dependent var
-
-                if (abs(xdif) <= abs(zdif))
-                {
-                    slope = xdif / zdif;
-
-                    float angle = (float)atan((double)1) * 2 - atan(slope);
-
-                    float zoffset   = cos(angle) / 2;
-                    float xoffset   = sin(angle) / 2;
-                    float maxXpoint = mobX + xoffset;
-                    float maxZpoint = mobZ - zoffset;
-                    float minXpoint = mobX - xoffset;
-                    float minZpoint = mobZ + zoffset;
+                // using same variable names as trick attack function, for consistent understanding
+                uint8 angleTAmob = worldAngle(PMob->loc.p, PCoverAbilityUser->loc.p);
+                float distTAmob  = distanceSquared(PCoverAbilityUser->loc.p, PMob->loc.p);
 
-                    maxSlope = ((maxXpoint - coverAbilityTargetX) / (maxZpoint - coverAbilityTargetZ));
-                    minSlope = ((minXpoint - coverAbilityTargetX) / (minZpoint - coverAbilityTargetZ));
-
-                    zDependent = false;
-                }
-                else
+                if (distTAmob <= static_cast<float>(PMob->GetMeleeRange() * PMob->GetMeleeRange()) && // make sure cover user is within melee range
+                    distTAmob >= 0.25f &&                                                             // require closer target not be closer than .5 yalms (.5*.5=.25 distsquared) to mob
```

Comment:
There's a couple of magic numbers floating (hah) around here, can you extract them into an anonymous namespace at the top of the file please? Here's an example from the top of zone_entities.cpp:
```cpp
namespace
{
    const float CHARACTER_SYNC_DISTANCE                = 45.0f;
    const float CHARACTER_DESPAWN_DISTANCE             = 50.0f;
    const int   CHARACTER_SWAP_MAX                     = 5;
    const int   CHARACTER_SYNC_LIMIT_MAX               = 32;
    const int   CHARACTER_SYNC_DISTANCE_SWAP_THRESHOLD = 30;
    const int   CHARACTER_SYNC_PARTY_SIGNIFICANCE      = 100000;
    const int   CHARACTER_SYNC_ALLI_SIGNIFICANCE       = 10000;
    const int   PERSIST_CHECK_CHARACTERS               = 20;
} // namespace
```

Reply:
Cool idea. I'll take a look